openpype/hosts/max/plugins/publish/extract_tycache.py

```diff
@@ -0,0 +1,186 @@
+
+        # Get the tyMesh filename for extraction
+        mesh_filename = f"{instance.name}__tyMesh.tyc"
+        mesh_repres = {
+            'name': 'tyMesh',
+            'ext': 'tyc',
+            'files': mesh_filename,
+            "stagingDir": stagingdir,
+            "outputName": '__tyMesh'
+        }
+        representations.append(mesh_repres)
+        self.log.info(
+            f"Extracted instance '{instance.name}' to: {mesh_filename}")
+
+    def get_files(self, instance, start_frame, end_frame):
+        """Get file names for tyFlow in tyCache format.
+
+        Set the filenames accordingly to the tyCache file
+        naming extension(.tyc) for the publishing purpose
+
+        Actual File Output from tyFlow in tyCache format:
+        <InstanceName>__tyPart_<frame>.tyc
+
+        e.g. tycacheMain__tyPart_00000.tyc
+
+        Args:
+            instance (pyblish.api.Instance): instance.
+            start_frame (int): Start frame.
+            end_frame (int): End frame.
+
+        Returns:
+            filenames(list): list of filenames
+
+        """
+        filenames = []
+        # should we include frame 0 ?
+        for frame in range(int(start_frame), int(end_frame) + 1):
+            filename = f"{instance.name}__tyPart_{frame:05}.tyc"
+            filenames.append(filename)
+        return filenames
+
+    def get_export_particles_job_args(self, members, start, end,
+                                      filepath, additional_attributes):
+        """Sets up all job arguments for attributes.
+
+        Those attributes are to be exported in MAX Script.
+
+        Args:
+            members (list): Member nodes of the instance.
+            start (int): Start frame.
+            end (int): End frame.
+            filepath (str): Output path of the TyCache file.
+            additional_attributes (dict): channel attributes data
+                which needed to be exported
+
+        Returns:
+            list of arguments for MAX Script.
+
+        """
+        job_args = []
+        opt_list = self.get_operators(members)
+        for operator in opt_list:
+            job_args.append(f"{operator}.exportMode=2")
+            start_frame = f"{operator}.frameStart={start}"
+            job_args.append(start_frame)
+            end_frame = f"{operator}.frameEnd={end}"
+            job_args.append(end_frame)
+            filepath = filepath.replace("\\", "/")
+            tycache_filename = f'{operator}.tyCacheFilename="{filepath}"'
+            job_args.append(tycache_filename)
+            # TODO: add the additional job args for tycache attributes
+            if additional_attributes:
+                additional_args = self.get_additional_attribute_args(
+                    operator, additional_attributes
+                )
+                job_args.extend(additional_args)
+            tycache_export = f"{operator}.exportTyCache()"
+            job_args.append(tycache_export)
+
+        return job_args
```

Comment:
I wonder if this is more readable.
```suggestion
        settings = {
            "exportMode": 2,
            "frameStart": start,
            "frameEnd": end,
            "tyCacheFilename": filepath.replace("\\", "/")
        }
        settings.update(additional_attributes)
        
        job_args = []
        for operator in self.get_operators(members):
            for key, value in settings.items():
                if isinstance(value, str):
                    # embed in quotes
                    value = f'"{value}"'
            
                job_args.append(f"{operator}.{key}=[value}")

            job_args.append(f"{operator}.exportTyCache()")

        return job_args
```

Note that I've removed the `bool` conversion to always be `True` . I assumed it'd be weird that if for whatever reason there's a bool value in additional attributes like: `key=False` that you'd still be forcing it `key=True` in the job args in the `get_additional_attribute_args()` function. Seemed like a bug.

Reply:
Why you are removing it? The tyflow channel data needs to be set to True in order to get the data export.

Reply:
It's not a bug, it's just hardcoded.

Reply:
Could you provide an example of the input data where that is not a bug? :)

Say `additional_attributes` is:
```python
additional_attributes = {
    "key1": True,
    "key2": False,
    "key3": True
}
```

I don't ever see a reason to set key2 to True? :p